OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -111,6 +111,13 @@ void RescanForTargets(Actor self)
 
 			while (!self.World.Map.Contains(moveTo) || !mobile.CanEnterCell(moveTo, null, false))
 			{
+				// without this check, this while can be infinity loop
+				if (moveTo == self.Location)
```

Comment:
How can the sandworm ever end up at a location that is either not in the map, or that it cannot enter?